File: algorithms/Trees/Easy/BinarySearchTreeSearching/VSDBinarySearchTreeSearching.py
class Node:
    def __init__(self,data):
        self.data=data 
        self.left=self.right=None 



def insertBST(root,data):

    if root is None:
    
        return Node(data)
    if root.data==data:
        return root 
    elif data<root.data:
        root.left=insertBST(root.left,data)
    else:
        root.right=insertBST(root.right,data)
    return root


def searchBST(root,data):
    if not root:
        return -1
    if root.data==data:
        return root
    elif data<root.data:
        return searchBST(root.left,data)
    elif data>root.data:
        return searchBST(root.right,data)


def inorder(root):
    if root:
        inorder(root.left)
        print(root.data,end=" ")
        inorder(root.right)


if __name__=="__main__":
    arr=[50,20,30,70,40,10]
    root=None 
    for i in arr:
        root=insertBST(root,i)
    # inorder(root)
    if searchBST(root,80):
        print("Found")
    else:
        print("Not Found")

File: algorithms/Trees/Easy/BinarySearchTreeSearching/VSDBinarySearchTreeSearching.java

class Node{
    int data;
    Node left,right;
    Node(int data){
        this.data=data;
    }
}

public class Main{

    static Node insertBST(Node root,int data){
        if(root==null) return new Node(data);
        if(root.data==data) return root;
        else if(data<root.data) root.left=insertBST(root.left,data);
        else root.right=insertBST(root.right,data);
        return root;
    }
    static void inorder(Node root){
        if(root!=null){
            inorder(root.left);
            System.out.print(root.data+" ");
            inorder(root.right);
        }
    }
    
    static Node searchBST(Node root,int data){
        if(root==null) return null;
        if(root.data==data) return root;
        else if(data<root.data) return searchBST(root.left, data);
        else return searchBST(root.right, data);
    }
    public static void main(String[] args) {
        int[] arr={50,20,30,70,40,10};
        Node root=null;
        for(int i:arr){
            root=insertBST(root,i);
        }
        if(searchBST(root, 80)!=null) System.out.println("Found");
        else System.out.println("Not Found");        
    }
}